### PR TITLE
Small improvements on `HiveColumnProjectionInfo`

### PR DESCRIPTION
**Pull request by findinpath**
## Description

Inspired from the changes in the PR https://github.com/trinodb/trino/pull/17085

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by krvikash — APPROVED** [review]

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java [hunk 1]
```diff
@@ -109,9 +108,7 @@ public boolean equals(Object obj)
 
     public static String generatePartialName(List<String> dereferenceNames)
     {
-        return dereferenceNames.stream()
-                .map(name -> "#" + name)
-                .collect(Collectors.joining());
+        return String.join("#", dereferenceNames);
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java:111]: This change returns a different result from the existing method. The previous method had a leading `#`. Are you changing it intentionally?

Comment (findinpath) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java:111]: Good catch. No, the change was not done intentionally.

Comment (ebyhr) [reply plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java:111]: CI failure relates to this change. Could you remove the commit from PR?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java [hunk 2]
```diff
@@ -123,4 +123,10 @@ public long getRetainedSizeInBytes()
                 // type is not accounted for as the instances are cached (by TypeRegistry) and shared
                 + estimatedSizeOf(partialName);
     }
+
+    @Override
+    public String toString()
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java:128]: nit: Move to under `equals` method.

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java [hunk 3]
```diff
@@ -123,4 +123,10 @@ public long getRetainedSizeInBytes()
...
+    {
+        return getPartialName() + ":" + type.getDisplayName();
```
Comment (ebyhr) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveColumnProjectionInfo.java:130]: nit: 
```suggestion
        return partialName + ":" + type.getDisplayName();
```

**Review by ebyhr — APPROVED** [review]

**Pull request merged**